### PR TITLE
Fix where the userCount is in the message

### PR DESCRIPTION
**Pull request by JustinKuli**
The message was changed, but the code which parses the message later
wasn't updated. Now that location is in a const which should make it
more searchable in the future.

Signed-off-by: Justin Kulikauskas <jkulikau@redhat.com>

I'm surprised there isn't a unit test for this, but the problem here did appear in the e2e tests in the framework repo. My PR there to update other failing things is blocked by this.

### CODE REVIEW

File: pkg/controller/iampolicy/iampolicy_controller.go [hunk 1]
```diff
@@ -323,8 +325,8 @@ func addViolationCount(plc *policiesv1.IamPolicy, roleName string, userCount int
 		return changed
 	}
 	firstNum := strings.Split(plc.Status.CompliancyDetails[plc.Name][namespace][0], " ")
-	if len(firstNum) >= 7 {
-		if firstNum[7] == fmt.Sprint(userCount) {
+	if len(firstNum) > idxOfUserCount {
```
Comment (mprahl) [inline pkg/controller/iampolicy/iampolicy_controller.go:328]: This isn't a safe way to parse the number since a role name can have a space (example below). Could you please use regex instead or take into account if there are spaces in `roleName`?

Here is an example using regex:
```go
package main

import (
	"fmt"
	"regexp"
	"strconv"
)

func main() {
	roleName := "example role"
	regexStr := `^(?:The number of users with the ` + regexp.QuoteMeta(roleName) + ` role is )(\d+)(?: above the specified limit)$`
	re := regexp.MustCompile(regexStr)
	
	status := "The number of users with the example role role is 5 above the specified limit"
	rv := re.FindStringSubmatch(status)
	numUsersStr := rv[1]
	if numUsersStr != "" {
		numUsers, _ := strconv.Atoi(numUsersStr)
		fmt.Printf("%d\n", numUsers)
	}
}
```

```yaml
~ ❯ oc get clusterrolebinding example -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
...
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: example role
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: User
  name: mprahl
```

File: pkg/controller/iampolicy/iampolicy_controller.go [hunk 2]
```diff
@@ -319,21 +323,21 @@ func addViolationCount(plc *policiesv1.IamPolicy, roleName string, userCount int
...
 	}
-	firstNum := strings.Split(plc.Status.CompliancyDetails[plc.Name][namespace][0], " ")
-	if len(firstNum) >= 7 {
-		if firstNum[7] == fmt.Sprint(userCount) {
-			return changed
+	regexStr := fmt.Sprintf(violationMsgFUserCountRegex, regexp.QuoteMeta(roleName))
+	re := regexp.MustCompile(regexStr)
+	regexGroups := re.FindStringSubmatch(plc.Status.CompliancyDetails[plc.Name][namespace][0])
+	if len(regexGroups) > 1 {
+		if regexGroups[1] == fmt.Sprint(userCount) {
```
Comment (mprahl) [inline pkg/controller/iampolicy/iampolicy_controller.go:332]: Optional:
```suggestion
	if len(regexGroups) > 1 && regexGroups[1] == fmt.Sprint(userCount) {
```

File: pkg/controller/iampolicy/iampolicy_controller.go [hunk 3]
```diff
@@ -346,9 +350,11 @@ func checkComplianceBasedOnDetails(plc *policiesv1.IamPolicy) {
 	}
 	for namespace, msgList := range plc.Status.CompliancyDetails[plc.Name] {
 		if len(msgList) > 0 {
-			violationNum := strings.Split(plc.Status.CompliancyDetails[plc.Name][namespace][0], " ")
-			if len(violationNum) >= 7 {
-				if violationNum[7] != fmt.Sprint(0) && strings.HasPrefix(violationNum[0], "Number") {
+			regexStr := fmt.Sprintf(violationMsgFUserCountRegex, regexp.QuoteMeta(roleName))
```
Comment (mprahl) [inline pkg/controller/iampolicy/iampolicy_controller.go:353]: Optional: It'd be nice to put this code in a helper function to avoid the duplication.

File: pkg/controller/iampolicy/iampolicy_controller.go [hunk 4]
```diff
@@ -346,9 +350,11 @@ func checkComplianceBasedOnDetails(plc *policiesv1.IamPolicy) {
...
+			re := regexp.MustCompile(regexStr)
+			regexGroups := re.FindStringSubmatch(plc.Status.CompliancyDetails[plc.Name][namespace][0])
+			if len(regexGroups) > 1 {
+				if regexGroups[1] != fmt.Sprint(0) {
```
Comment (mprahl) [inline pkg/controller/iampolicy/iampolicy_controller.go:357]: Optional:
```suggestion
			if len(regexGroups) > 1 && regexGroups[1] != fmt.Sprint(0) {
```

**Comment by JustinKuli** [comment]
I think these refactorings are good, although I think they also expose some more cases where maybe the logic wasn't perfect before wrt error handling and edge cases. But I really kind of need this merged sooner so that I can unblock the canary tests.

That said, I don't want to rush this. Please look over the newest update, and if it doesn't break anything *new*, I would like to merge it.

**Review by mprahl — APPROVED** [review]
Looks great!

**Comment by JustinKuli** [comment]
/retest

plz

**Comment by JustinKuli** [comment]
```
STEP 7: RUN make build
.build-harness-bootstrap:1: *** multiple target patterns.  Stop.
```

The build harness is so good until it isn't! I feel like something got messed up in a previous image layer or something, maybe when it was trying to curl the build harness stuff? :(

/retest-required

**Review by mprahl — APPROVED** [review]

**Comment by JustinKuli** [comment]
I'll wait another 30 minutes I guess

**Comment by dhaiducek** [comment]
Wait a little bit and then put another minor commit in. That seemed to work for me.

**Comment by JustinKuli** [comment]
Will do

**Review by mprahl — APPROVED** [review]
